kernel/CMakeLists.txt

```diff
@@ -42,6 +42,10 @@ target_sources_ifdef(CONFIG_ATOMIC_OPERATIONS_C   kernel PRIVATE atomic_c.c)
 target_sources_ifdef(CONFIG_MMU                   kernel PRIVATE mmu.c)
 target_sources_ifdef(CONFIG_POLL                  kernel PRIVATE poll.c)
 
+if(NOT CONFIG_MULTITHREADING)
+  message(WARNING "CONFIG_MULTITHREADING is deprecated")
```

Comment:
I think we need to be a little more clear.  This could be a little confusing (I know its only going to trigger of CONFIG_MULTITHREADING=n).

Reply:
Do we need to do that though? it defaults to `y` and we are going to remove the entire Kconfig option after 2 releases, we are not going to "freeze it to `y`"?

Reply:
I understand, just saying that if I read that the code w/o thinking too much it reads as if MULTITHREADING feature is going away or deprecated.

Reply:
We can just make it:

`message(WARNING "CONFIG_MULTITHREADING=n is deprecated")`

Reply:
@galak's suggestion is fine to me

Reply:
Thinking about this, I believe the option should have been called `config NO_MULTITHREADING`

Reply:
How about:
```
  message(WARNING "Single threading mode has been selected.\n Single threading mode is deprecated (CONFIG_MULTITHREADING=n)")
```

> Do we need to do that though? it defaults to y and we are going to remove the entire Kconfig option after 2 releases, we are not going to "freeze it to y"?

That depends on what gives the best user experience.
If we remove it, then users that has `CONFIG_MULTITHREADING=n` in a conf file, will see:
```
/projects/github/ncs/zephyr/samples/hello_world/prj.conf:2: warning: attempt to assign the value 'n'
 to the undefined symbol MULTITHREADING
```

but if we remove the prompt, and thus force it to `y`, then they will see:
```
error: MULTITHREADING (defined at kernel/Kconfig:12) is assigned in a configuration file, but is not
directly user-configurable (has no prompt). It gets its value indirectly from other symbols. See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_MULTITHREADING.html and/or look up
MULTITHREADING in the menuconfig/guiconfig interface.
```
Which re-directs them to http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_MULTITHREADING.html
where it can be described that this setting has been removed, and why,.

A much better user experience, than the: `attempt to assign the value 'n' to the undefined symbol MULTITHREADING` message, imo.